[datadog_user] Fix the `name` field as Optional + Computed (#2784)
DataDog/terraform-provider-datadog: docs/resources/user.md

### Optional

- `disabled` (Boolean) Whether the user is disabled. Defaults to `false`.

> - `name` (String) Name for user. Should be set only in case of password authentication as it is overriden from Google or SAML authentication.

Reviewer: ```suggestion
- `name` (String) User's name. Should be set only for password authentication, as it is overridden by Google or SAML authentication.
```
